Vagrantfile

```diff
@@ -0,0 +1,32 @@
+Vagrant.configure("2") do |config|
+  config.vm.box = "hashicorp/bionic64"
+
+  config.vm.provision "shell", privileged: false, inline: <<-SHELL
+    sudo apt-get update
+	sudo apt-get install -y build-essential libssl-dev zlib1g-dev libbz2-dev libreadline-dev libsqlite3-dev wget curl llvm libncurses5-dev libncursesw5-dev xz-utils tk-dev libffi-dev liblzma-dev python-openssl git
+	curl https://pyenv.run | bash
+	echo 'export PATH="$HOME/.pyenv/bin:$PATH"' >> ~/.profile
+	echo 'eval "$(pyenv init -)"' >> ~/.profile
+	echo 'eval "$(pyenv virtualenv-init -)"' >> ~/.profile
+	source ~/.profile
+	export PATH="/home/vagrant/.pyenv/bin:$PATH"
+	eval "$(pyenv init -)"
+	eval "$(pyenv virtualenv-init -)"
+	pyenv install 3.8.5
+	pyenv global 3.8.5
+	curl -sSL https://raw.githubusercontent.com/python-poetry/poetry/master/get-poetry.py | python
+  SHELL
+  
+  config.trigger.after :up do |trigger|
+    trigger.name = "Launching App"
+    trigger.info = "Running the TODO app setup script"
+    trigger.run_remote = {privileged: false, inline: "
+      cd /vagrant
+      poetry install
+      poetry run flask run --host 0.0.0.0
```

Comment:
This works as much as the exercise brief asked for but you may have noticed it means `vagrant up` doesn't complete and then vagrant manager gets confused and you can't easily run e.g. `vagrant halt` afterwards

You should run the web app as a background process. E.g. 
`nohup poetry run flask run --host 0.0.0.0 > logs.txt 2>&1 &` where `nohup ... &` means to run the command as a separate process. `> logs.txt 2>&1` is to save the output to a file.